api/query.py

```diff
@@ -40,13 +50,45 @@ class Position(NamedTuple):
     line: int
     column: int
 
+    def __eq__(self, other) -> bool:
+        return self.line == other.line and other.column == self.column
+
+    def __gt__(self, other) -> bool:
+        return self.line > other.line or (
+            other.line == self.line and other.column <= self.column
```

Comment:
Consistency makes it sooo much easier for the reader to understand code. So I'd suggest:

1. Always keeping to operands in the comparison in the same order (eg. `self` on left, `other` on right)
2. Since this is a implementation of `>`, use `>` in the comparisons rather than `<=`
Apply to the `__lt__` too

```suggestion
            self.line == other.line  and self.column > other.column
```